packages/fluentui/react-bindings/src/compose/useCompose.ts

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+
+import { COMPOSE_PROP } from './compose';
+import { ComposePreparedOptions, ComposeResolvedOptions } from './types';
+import useReactElement from './useReactElement';
+
+function useCompose(): ComposeResolvedOptions | null {
+  const [ElementType, props = {}] = useReactElement<
+    React.FunctionComponent & { [COMPOSE_PROP]: ComposePreparedOptions | undefined },
+    Record<string, any>
+  >();
+  const composeOptions: ComposePreparedOptions | undefined = ElementType?.[COMPOSE_PROP];
+
+  if (composeOptions) {
+    const { className, displayNames, handledProps, mapPropsToBehaviorChain, mapPropsToStylesChain, overrideStyles } = composeOptions;
+
+    return {
+      className,
+      displayNames,
+
+      behaviorProps: mapPropsToBehaviorChain.reduce((acc, fn) => ({ ...acc, ...fn(props) }), {}),
+      stylesProps: mapPropsToStylesChain.reduce((acc, fn) => ({ ...acc, ...fn(props) }), {}),
+
+      handledProps,
+      overrideStyles
+    };
+  }
+
+  return null;
```

Comment:
```suggestion
  return ElementType?.[COMPOSE_PROP] || null;
```

As `behaviorProps` & `stylesProps` should be resolved in hooks with proper state